router/transformer/transformer_test.go

```diff
@@ -0,0 +1,185 @@
+		{JobMetadataArray: []types.JobMetadataT{{JobID: 2}}, StatusCode: http.StatusInternalServerError, Error: expectedErrorTxt},
+		{JobMetadataArray: []types.JobMetadataT{{JobID: 3}}, StatusCode: http.StatusInternalServerError, Error: expectedErrorTxt},
+	}
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Add(apiVersionHeader, strconv.Itoa(utilTypes.SUPPORTED_TRANSFORMER_API_VERSION))
+		_, err := w.Write([]byte("invalid json"))
+		require.NoError(t, err)
+	}))
+	defer svr.Close()
+	t.Setenv("DEST_TRANSFORM_URL", svr.URL)
+	tr := NewTransformer()
+
+	transformMessage := types.TransformMessageT{
+		Data: []types.RouterJobT{
+			{JobMetadata: types.JobMetadataT{JobID: 1}},
+			{JobMetadata: types.JobMetadataT{JobID: 2}},
+			{JobMetadata: types.JobMetadataT{JobID: 3}},
+		},
+	}
+	transformerResponse := tr.Transform(BATCH, &transformMessage)
+	normalizeErrors(transformerResponse, expectedErrorTxt)
+	require.NotNil(t, transformerResponse)
+	require.Equal(t, expectedTransformerResponse, transformerResponse)
+}
+
+func TestTransformValidationInOutMismatchError(t *testing.T) {
+	initMocks(t)
+	config.Load()
+	pkgLogger = logger.NOP{}
+	expectedErrorTxt := "Transformer returned invalid output size: 4 for input size: 3"
+	expectedTransformerResponse := []types.DestinationJobT{
+		{JobMetadataArray: []types.JobMetadataT{{JobID: 1}}, StatusCode: http.StatusInternalServerError, Error: expectedErrorTxt},
+		{JobMetadataArray: []types.JobMetadataT{{JobID: 2}}, StatusCode: http.StatusInternalServerError, Error: expectedErrorTxt},
+		{JobMetadataArray: []types.JobMetadataT{{JobID: 3}}, StatusCode: http.StatusInternalServerError, Error: expectedErrorTxt},
+	}
+	serverResponse := []types.DestinationJobT{
+		{JobMetadataArray: []types.JobMetadataT{{JobID: 1}}},
+		{JobMetadataArray: []types.JobMetadataT{{JobID: 2}}},
+		{JobMetadataArray: []types.JobMetadataT{{JobID: 3}}},
+		{JobMetadataArray: []types.JobMetadataT{{JobID: 4}}},
+	}
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Add(apiVersionHeader, strconv.Itoa(utilTypes.SUPPORTED_TRANSFORMER_API_VERSION))
+		b, err := json.Marshal(serverResponse)
+		require.NoError(t, err)
+		_, err = w.Write(b)
+		require.NoError(t, err)
+	}))
+	defer svr.Close()
+	t.Setenv("DEST_TRANSFORM_URL", svr.URL)
+	tr := NewTransformer()
+
+	transformMessage := types.TransformMessageT{
+		Data: []types.RouterJobT{
+			{JobMetadata: types.JobMetadataT{JobID: 1}},
+			{JobMetadata: types.JobMetadataT{JobID: 2}},
+			{JobMetadata: types.JobMetadataT{JobID: 3}},
+		},
+	}
+	transformerResponse := tr.Transform(BATCH, &transformMessage)
+	normalizeErrors(transformerResponse, expectedErrorTxt)
+	require.NotNil(t, transformerResponse)
+	require.Equal(t, expectedTransformerResponse, transformerResponse)
+}
+
+func TestTransformValidationJobIDMismatchError(t *testing.T) {
```

Comment:
[minor/suggestion] You could have used table-driven tests here, to reduce the setup overhead (although small in this case) and make it easier to write multiple scenarios.

Great job writing those tests.

Reply:
I find it more readable this way & the overhead is not too much (IMO)